pkg/network/ebpf/c/protocols/kafka/kafka-parsing.h

```diff
@@ -112,6 +118,7 @@ static __always_inline bool kafka_process(kafka_transaction_t *kafka_transaction
     if (topic_name_size <= 0 || topic_name_size > TOPIC_NAME_MAX_ALLOWED_SIZE) {
         return false;
     }
+    update_topic_name_size_telemetry(kafka_tel, topic_name_size);
```

Comment:
That should be moved to line 117, ad the if clause will "remove" every topic name that is longer than we support, and we won't know if the upper limit is sufficient

Reply:
Since the maximum topic name size in Kafka is [255](https://www.google.com/search?q=kafka+maximum+topic+name+length&rlz=1C5GCCM_en&oq=kafka+maximum+topic+name+length&gs_lcrp=EgZjaHJvbWUyBggAEEUYOTIICAEQABgWGB4yDQgCEAAYhgMYgAQYigUyDQgDEAAYhgMYgAQYigUyDQgEEAAYhgMYgAQYigUyBggFEEUYQDIGCAYQRRhA0gEIMzc5NGowajeoAgCwAgA&sourceid=chrome&ie=UTF-8), there's no need to check for that, as packets with such size are not Kafka traffic. Additionally, we won't encounter these packets in the parsing process because they will be dropped during the classification phase